### DISCUSSION TITLE
Timed shutdown, auto restore connection

### DISCUSSION THREAD

**Discussion by cbturner46** [nutdotnet/WinNUT-Client#158] [Ideas]
New user here and I think I see some of this in fixes to the source code but not in the ready-to-roll binary release.
I think I'd like to have the option to shutdown my PC after a set time on battery as opposed to percentage of energy remaining.  Reason is that the NAS shuts down after a configured time and you need to ensure the PC shuts down first whilst the NAS is still serving NUT (I think).
Also, an option to auto reconnect once WinNUT is started on the PC by trying to contact the NAS periodically until its service starts.
Otherwise it's all good and many thanks to gbakeman for all the great work and for pointing me at this new repository.  Need to kill those old google searches somehow as that's how I found my way to WinNUT but it was to the old (non working) releases.

**Comment by gbakeman** [comment]
Hey @cbturner46,

Thanks for testing the newer releases and posting back here. I know a few other people have requested a timer-based shutdown which we don't really do yet. I'd encourage you to follow #92 where I'll probably implement that there. I also think #65 will address your idea about attempting connections indefinitely. I'm also curious, what search(es) led you to the original WinNUT repository? I'm tracking our visibility in #91, if you have any further thoughts about boosting us I'd appreciate that there!

**Reply by cbturner46** [reply]
Thinking about the timer-based shutdown, may I suggest that my use case is a common one.  Rather than having the PC last as long as possible on UPS battery, I want to shut it down safely after just a few minutes once it's probable that the power will be off for a while.  Then shut down the NAS so timer-based shutdown is required to to both in the right order.  Thereafter, the UPS battery is preserved for as long as possible to power the broadband router for Wi-Fi and digital voice service, plus acting as local Ethernet switch.

**Reply by gbakeman** [reply]
Thanks again for making that post in the Synology forums. With regards to your suggestion, I totally agree that it's a sensible idea that ought to be implemented. You can sort of achieve this in the _Shutdown Options_ menu by setting the **Shutdown if battery lower than __ %** value to 100, then adjusting the **Delay to Shutdown (sec)** field to however long you'd like for WinNUT to wait while on battery power before shutting down the PC. Regarding shutting down the NAS though, that isn't really something WinNUT can handle at the moment and I don't think it will be able to if the host PC is already shutdown. I did link to this discussion in #92 where I hope to tackle the ideas at hand, and if I do end up implementing #92, then you should be able to execute a custom script which will tell the NAS to power off before the PC does.

**Comment by cbturner46** [comment]
I think most of the misleading stuff was in closed threads on Synology's forums.  I’ll put a new post there tomorrow which may help.

From my iPad
***@***.***
________________________________

**Reply by cbturner46** [reply]
Done. See https://community.synology.com/enu/forum/1/post/188123

**Comment by cbturner46** [comment]
Yes, that might work, although I need to check if my APC UPS dips below 100% occasionally even when there's no outage.  Not sure about your second point as I wasn't suggesting that the Win client should be able to shut down the NAS.  My Synology NAS has timed shutdown capability already.

Cheers, Chris
________________________________

**Reply by gbakeman** [reply]
> Not sure about your second point as I wasn't suggesting that the Win client should be able to shut down the NAS.  My Synology NAS has timed shutdown capability already.

Oh, I understand! Yeah that should be fine then. I also want to mention that the shutdown conditions aren't evaluated until the UPS indicates that it's on battery, so effectively you're guaranteeing that the countdown begins immediately once mains power is lost and battery power is engaged. Although I want to say that this logic isn't very clear in the current iteration so I do want to implement your idea. I'm currently working on the gauges in WinNUT and hope to address a few pending issues with those, then I may try finalizing the WinNUT upgrade process before tackling this.

**Comment by cbturner46** [comment]
Many thanks for the clarification.

Cheers, Chris
from my phone
________________________________
